strapi/strapi
# fix(database): Add groupBy after distinct queries (option 2)

Alternative solution: https://github.com/strapi/strapi/pull/20124

### What does it do?

After performing a query that is supposed to be `distinct` but should actually probably be a `groupBy`, we apply the same groupBy on the resultset received, in Node rather than the db query.

NOTE: on further review and discussion, this will probably break pagination with a resultset that exceeds the size of one page.

### Why is it needed?

Fixes https://github.com/strapi/strapi/issues/11892

### How to test it?

See [the original issue](https://github.com/strapi/strapi/issues/11892).

A test of `sort` has been added to prevent regressions, but this may have unintended side effects that are not caught by existing tests, so a pretty thorough test of nested populates and queries with joins and distincts that aren't the typical path we use.

### Related issue(s)/PR(s)

Alternative solution: https://github.com/strapi/strapi/pull/20124

Fixes https://github.com/strapi/strapi/issues/11892

DX-117
